mongodbforms/fields.py

```diff
@@ -116,26 +119,31 @@ def clean(self, value):
             else:
                 return None
 
-        oid = super(ReferenceField, self).clean(value)
+        return self.queryset.get(pk=super(ReferenceField, self).clean(value))
 
+    def valid_value(self, value):
```

Comment:
because the original code is calling `super().clean(value)` this actually means it checks each value rather than doing the more optimal DB check (see `ChoiceField.valid_value`). In order to utilize the parent's code path more effectively (least surprises), the `valid_value` function does the validation, and `validate` handles throwing the exception. Because the value isn't passed out of clean I make a second DB call to get the value (if it's valid). It's not exactly ideal but the call should be O(1) as it's a field that will be indexed.